packages/amplify_api/example/integration_test/graph_ql_tests.dart

```diff
@@ -44,12 +103,94 @@ void main() {
           }
         }
       }''';
-
       var _r = await Amplify.API
           .query<String>(request: GraphQLRequest(document: graphQLDocument));
       var response = await _r.response;
       Map data = jsonDecode(response.data);
       expect(data[listBlogs][items], hasLength(greaterThanOrEqualTo(0)));
     });
+
+    // Queries
+    testWidgets('should GET a blog with Model helper',
+        (WidgetTester tester) async {
+      String name = "Integration Test Blog to fetch";
+      Blog blog = await addBlog(name);
+
+      var req = ModelQueries.get(Blog.classType, blog.id);
+      var _r = await Amplify.API.query(request: req);
+
+      var res = await _r.response;
+      Blog data = res.data;
+
+      expect(data.name, name);
+    });
+
+    testWidgets('should LIST blogs with Model helper',
+        (WidgetTester tester) async {
+      String blog_1_name = "Integration Test Blog 1";
+      String blog_2_name = "Integration Test Blog 2";
+      String blog_3_name = "Integration Test Blog 3";
+      Blog blog_1 = await addBlog(blog_1_name);
+      Blog blog_2 = await addBlog(blog_2_name);
+      Blog blog_3 = await addBlog(blog_3_name);
+
+      var req = ModelQueries.list<Blog>(Blog.classType);
+      var _r = await Amplify.API.query(request: req);
+
+      var res = await _r.response;
+      var data = res.data;
+
+      expect(data.items.length, greaterThanOrEqualTo(3));
+    });
+
+    // Mutations
+    testWidgets('should CREATE a blog with Model helper',
+        (WidgetTester tester) async {
+      String name = "Integration Test Blog - create";
+      Blog blog = Blog(name: name);
+
+      blog = blog.copyWith(name: name);
+      var req = ModelMutations.create(blog);
+
+      var _r = await Amplify.API.mutate(request: req);
+
+      var res = await _r.response;
+      Blog data = res.data;
+
+      blogIds.add(data);
+
+      expect(data.name, name);
+    });
+
+    testWidgets('should UPDATE a blog with Model helper',
+        (WidgetTester tester) async {
+      String oldName = "Integration Test Blog to update";
+      String newName = "Integration Test Blog - updated";
+      Blog blog = await addBlog(oldName);
+
+      blog = blog.copyWith(name: newName);
+
+      var req = ModelMutations.update(blog);
+      var _r = await Amplify.API.mutate(request: req);
+
+      var res = await _r.response;
+      Blog data = res.data;
+
+      expect(data.name, newName);
+    });
+
+    testWidgets('should DELETE a blog with Model helper',
+        (WidgetTester tester) async {
+      String name = "Integration Test Blog - delete";
+      Blog blog = await addBlog(name);
+
+      var req = ModelMutations.delete(blog);
+      var _r = await Amplify.API.mutate(request: req);
+
+      var res = await _r.response;
+      Blog data = res.data;
+
+      expect(data.name, name);
+    });
```

Comment:
suggestion: query the list of blogs before and after the delete to ensure the blog was actually deleted. Easiest way is to check the count of the result, ensure it went down by 1. Alternatively, you could query the list after the delete, iterate over all the blogs and ensure the blog not in the list result.